### PR TITLE
Enforce formatting with CI

### PR DESCRIPTION
**Pull request by eulertour**
Formats the library code, examples, and tests with black and adds a CI check that new PRs are black compliant.

### CODE REVIEW

**Comment by eulertour** [comment]
This PR addresses formatting, but _linting_ the library will take some work, since it's currently riddled with pylint violations.

File: manim/mobject/svg/text_mobject.py [hunk 1]
```diff
@@ -96,22 +100,27 @@ def __init__(self, text, **config):
     def get_space_width(self):
         size = self.size * 10
 
-        dir_name = file_writer_config['text_dir']
-        file_name = os.path.join(dir_name, "space") + '.svg'
+        dir_name = config["text_dir"]
```
Comment (Aathish04) [inline manim/mobject/svg/text_mobject.py:103]:
```suggestion
        dir_name = file_writer_config["text_dir"]
```

This needs to be changed to use file_writer_config. It showed up as a merge conflict, and I tried to resolve it, but it ended up reversing all the formatting changes black made and I had to revert what I did. Sorry 'bout that.

File: manim/mobject/svg/text_mobject.py [hunk 2]
```diff
@@ -288,12 +309,13 @@ def text2svg(self):
...
         lsh = self.lsh * 10
 
-        if self.font == '':
-            if NOT_SETTING_FONT_MSG != '':
+        if self.font == "":
+            if NOT_SETTING_FONT_MSG != "":
                 logger.warning(NOT_SETTING_FONT_MSG)
-        dir_name = file_writer_config['text_dir']
+
+        dir_name = config["text_dir"]
```
Comment (Aathish04) [inline manim/mobject/svg/text_mobject.py:316]:
```suggestion
        dir_name = file_writer_config["text_dir"]
```

Same here. I think `file_writer_config` needs to be imported again as well...

File: tests/testing_utils.py [hunk 3]
```diff
@@ -34,19 +34,23 @@ class SceneTester:
...
-            file_writer_config['text_dir'] = os.path.join(
-                self.path_tests_medias_cache, scene_object.__name__, 'Text')
-            file_writer_config['tex_dir'] = os.path.join(
-                self.path_tests_medias_cache, scene_object.__name__, 'Tex')
-
-        file_writer_config['skip_animations'] = True
-        config['pixel_height'] = 480
-        config['pixel_width'] = 854
-        config['frame_rate'] = 15
+            config["text_dir"] = os.path.join(
```
Comment (Aathish04) [inline tests/testing_utils.py:43]:
```suggestion
            file_writer_config["text_dir"] = os.path.join(
```

**Comment by leotrs** [comment]
My only problem with this is that it will destroy `git blame`. I'm fine with that, but wanted to bring it up to make sure we are all fine with that. @ManimCommunity/core

**Comment by PgBiel** [comment]
> My only problem with this is that it will destroy `git blame`. I'm fine with that, but wanted to bring it up to make sure we are all fine with that. @ManimCommunity/core

not sure if this is a good idea then... we should just responsibilize the author to format using black by themselves?

**Comment by eulertour** [comment]
If we only format files gradually the formatting changes will pollute future diffs for a long time. It'd be best to just do it all at once and go backward two steps in the blame for a while.

**Comment by leotrs** [comment]
Hmm you raise a good point...

**Comment by PgBiel** [comment]
If we can make the blame remain functional then I’m all for it

**Comment by eulertour** [comment]
The blame will be changed no matter what. The only difference is in whether we do it at once or draw it out over months of commits while trying to selectively enforce formatting on only the files that we've already formatted. It'd be much easier to just look two steps backward in the blame when it's necessary.

**Pull request merged**